src/core/server/integration_tests/saved_objects/service/lib/update.test.ts

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import Path from 'path';
+import fs from 'fs/promises';
+import { pick } from 'lodash';
+import type { ElasticsearchClient } from '@kbn/core-elasticsearch-server';
+import type { SavedObjectsType, SavedObjectsModelVersionMap } from '@kbn/core-saved-objects-server';
+import { type TestElasticsearchUtils } from '@kbn/core-test-helpers-kbn-server';
+import '../../migrations/jest_matchers';
+import {
+  getKibanaMigratorTestKit,
+  startElasticsearch,
+} from '../../migrations/kibana_migrator_test_kit';
+import { delay } from '../../migrations/test_utils';
+import { getBaseMigratorParams } from '../../migrations/fixtures/zdt_base.fixtures';
+
+export const logFilePath = Path.join(__dirname, 'update.test.log');
+
+describe('SOR - update API', () => {
+  let esServer: TestElasticsearchUtils['es'];
```

Comment:
A basic integration test asserting the behavior when accessing / updating a SO from repositories having different model versions registered (simulating version cohabitation)